monitoring/reporter/register.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package reporter
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+//The register defines a global interface for managing the various metrics outputs
+//This might be overkill for what we have now, but I'd like to at least try to prepare
```

Comment:
Agreed that it would be preferable to not use the `config.Namespace` type. Not sure what you mean by `For this phase we're specifically scoped to a single output per executable` ? Do you mean a single queue monitoring output? I was kinda going off this assumption that the logging monitoring and `expvar` weren't mutually exclusive, but more "default always on" and "optional debugging interface"